tests/integration/test_pip_packages.py

```diff
@@ -0,0 +1,146 @@
+    repo = test_env["pip_packages"]["without_deps"]["repo"]
+    ref = test_env["pip_packages"]["without_deps"]["ref"]
+    pkg_managers = ["pip"]
+    initial_response = client.create_new_request(
+        payload={
+            "repo": repo,
+            "ref": ref,
+            "pkg_managers": pkg_managers,
+            # TODO: delete pip-dev-preview flag when
+            #  the pip package manager will be ready for production usage
+            "flags": ["pip-dev-preview"],
+        },
+    )
+    completed_response = client.wait_for_complete_request(initial_response)
+    assert completed_response.status == 200
+
+    expected_types = test_env["pip_packages"]["without_deps"]["expected_types"]
+    assert_packages_from_response(completed_response.data, expected_types)
+
+    assert_dependencies_from_response(completed_response.data)
+
+    # Download and extract source tarball
+    source_name = tmpdir.join(f"download_{str(completed_response.id)}")
+    file_name_tar = tmpdir.join(f"download_{str(completed_response.id)}.tar.gz")
+    resp = client.download_bundle(completed_response.id, file_name_tar)
+    assert resp.status == 200
+    assert tarfile.is_tarfile(file_name_tar)
+    with tarfile.open(file_name_tar, "r:gz") as tar:
+        tar.extractall(source_name)
+
+    expected_file_urls = test_env["pip_packages"]["without_deps"]["expected_files"]
+
+    # Check that the source tarball includes the application source code under the app directory.
+    assert_expected_files(path.join(source_name, "app"), expected_file_urls)
+    # Check that the source tarball includes an empty deps directory.
+    assert_expected_files(path.join(source_name, "deps"))
+
+    purl = test_env["pip_packages"]["without_deps"]["purl"]
+    image_contents = [{"dependencies": [], "purl": purl, "sources": []}]
+    assert_pip_content_manifest(client, completed_response.id, image_contents)
+
+
+def assert_packages_from_response(response_data, expected_types):
+    """
+    Check amount and type of packages in the response data.
+
+    :param dict response_data: response data from the Cachito request
+    :param list expected_types: expected types of packages
+    """
+    assert len(response_data["packages"]) == len(expected_types)
+    for idx in range(len(expected_types)):
+        assert response_data["packages"][idx]["type"] == expected_types[idx]
+
+
+def assert_dependencies_from_response(response_data, expected_deps=None):
+    """
+    Check that dependencies are correctly listed in the response data.
+
+    :param dict response_data: response data from the Cachito request
+    :param list expected_deps: expected package dependencies
+    """
+    if expected_deps is None:
+        expected_deps = []
+
+    assert set(response_data["dependencies"]) == set(expected_deps)
+
+    packages_deps = []
+    for package in response_data["packages"]:
+        packages_deps += package["dependencies"]
+    assert set(packages_deps) == set(expected_deps)
```

Comment:
Instead of checking that the union of all dependencies is what we expect, could we check that each package has the correct dependencies? i.e. if you have the following response:
```
{
  "packages": [
    {
      "name": "package-a"
      "dependencies": [A, B]
    },
    {
      "name": "package-b",
      "dependencies": [B, C]
    }
  ]
}
```
IMO it would be better to check that package-a has dependencies A and B,  package-b has B and C

Reply:
I just moved it to one assertion `assert_packages_from_response`. Thank you!